### ISSUE TITLE
Shipping Method for Multishipping orders isn't saved

### ISSUE THREAD

**Issue by atsareva** [magento/magento2#17363]
Hi guys,

I've found that all orders which have been created via multishipping checkout don't contain 'shipping_method' info.

### Preconditions
1. Magento 2.2.5 (with sample data)
2. Ubuntu 16.04 lts
3. PHP 7.0.30
4. mysql  Ver 14.14 Distrib 5.7.22
5. apache

### Steps to reproduce
1. Login in with - roni_cost@example.com / roni_cost3@example.com
2. Add some products to the cart
3. Go to multishipping checkout and create order.
4. Go to order view page

![review order magento commerce](https://user-images.githubusercontent.com/1243876/43674832-2fad097e-97e2-11e8-80c8-82893fb7c970.png)
![success page](https://user-images.githubusercontent.com/1243876/43674835-35538808-97e2-11e8-8c08-baea662cfa59.png)
![magento 2 2 5 loc sales order view order_id 11](https://user-images.githubusercontent.com/1243876/43674837-387dd4de-97e2-11e8-8efa-43f5b3b0a517.png)

### Expected result
I should have ability to go to order view page, but it doesn't work.

### Actual result
'shipping_method' column ('sales_order' table) should contain method code, but it's NULL. Also I get error when got to the order view page.

Thanks,
Alona Tsarova

**Comment by magento-engcom-team** [comment]
Hi @atsareva. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@atsareva do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by atsareva** [comment]
@magento-engcom-team give me 2.2.5 instance

**Comment by magento-engcom-team** [comment]
Hi @atsareva. Thank you for your request. I'm working on Magento 2.2.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @atsareva, here is your [Magento instance](https://i-17363-2-2-5.engcom.dev.magento.com).
Admin access: [https://i-17363-2-2-5.engcom.dev.magento.com/admin](https://i-17363-2-2-5.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by atsareva** [comment]
Hi @magento-engcom-team ,
I can't  reproduce this error properly on the test environment because of
1. env. should be in developer mode
2. server should be configured to show all errors and notices
3. I don't have access to the db, so I can't check 'shipping_method' column
4. I don't get this error on live project due to mentioned configurations, but the column contains NULL, so we have some problems system which should get shipping info.

Thanks,
Alona

**Comment by engcom-backlog-nickolas** [comment]
Hello @atsareva, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by mam08ixo** [comment]
I can confirm that the shipping method is no longer stored with multishipping orders since Magento 2.2.5 (0fc3d07).

With the _Temando Shipping_ module not installed, the exact same error can be observed when trying to access the _New Shipment_ page:

```
Exception #0 (Exception): Notice: Undefined offset: 1 in vendor/magento/module-sales/Model/Order.php on line 1205
#0 vendor/magento/module-sales/Model/Order.php(1205): Magento\Framework\App\ErrorHandler->handler(8, 'Undefined offse...', '...', 1205, Array)
#1 vendor/magento/module-shipping/Block/Adminhtml/Create/Items.php(150): Magento\Sales\Model\Order->getShippingMethod(true)
#2 vendor/magento/module-shipping/view/adminhtml/templates/create/items.phtml(70): Magento\Shipping\Block\Adminhtml\Create\Items->canCreateShippingLabel()
```

**Comment by tsn-Aleksey** [comment]
#khcd2018

**Comment by magento-engcom-team** [comment]
@tsn-Aleksey thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by tsn-Aleksey** [comment]
I can't to reproduce this bug in 2.2.5. But I found a duplicate of the current task (#17062)

**Comment by tsn-Aleksey** [comment]
I’ve found that issue was fixed in 2.2-develop in https://github.com/magento/magento2/commit/6366478
